### PR TITLE
Email notification clean branch

### PR DESCRIPTION
**Pull request by adiraju13**
Just need to fix styling, but it seems like all the functionality is there

### CODE REVIEW

**Comment by jerrylinew** [comment]
@adiraju13 can you resolve this conflict by merging master into your branch and resolving conflicts?

File: models/emailer.js [hunk 1]
```diff
@@ -46,6 +45,98 @@ exports.setup_potential_trade_email = function(user_id, owned_book, target_user,
 };
 
 
+exports.send_rejected_trade_email = function (trade_id, next) {
+    foundTrades.get_trade_by_id(trade_id, function(err, result) {
+        if (err) next(err);
+        for (var i = 0; i < result.length; i++) {
```
Comment (codyleyhan) [inline models/emailer.js:51]: change var to let

File: models/emailer.js [hunk 2]
```diff
@@ -46,6 +45,98 @@ exports.setup_potential_trade_email = function(user_id, owned_book, target_user,
...
+                    target_user_name: email_data.target_name,
+                    target_book_name: email_data.wanted_book_name,
+                    target_book_author: email_data.wanted_book_author,
+                    owned_book_name: email_data.have_book_name,
+                    owned_book_author: email_data.have_book_author
+                }
+                var template = handlebars.compile(rejected_email_html);
+                var custom_html = template(replacements);
+
+                sgMail.setApiKey(process.env.EMAIL_KEY);
```
Comment (codyleyhan) [inline models/emailer.js:65]: probably better to use some kind of dependency injection to add this stuff in, or better yet, use a closure.

```js
const make_send_rejected_trade_email = function(api_key, from_key) {
   func (trade_id, next)  {
     //  now has access to both variables
  }
}
```

File: models/emailer.js [hunk 3]
```diff
@@ -46,6 +45,98 @@ exports.setup_potential_trade_email = function(user_id, owned_book, target_user,
...
+};
+
+exports.send_accepted_trade_email = function (trade_id, next) {
+    foundTrades.get_trade_by_id(trade_id, function(err, result) {
+        if (err) next(err);
+        var emails = [];
+        var replacements = [];
+        logger.log(result);
+        var counter = 0;
+        for (var i = 0; i < result.length; i++) {
```
Comment (codyleyhan) [inline models/emailer.js:87]: style: Seems like you and jerry have different styles on space before curly braces or not

File: models/emailer.js [hunk 4]
```diff
@@ -46,6 +45,98 @@ exports.setup_potential_trade_email = function(user_id, owned_book, target_user,
...
+    for (var i = 0; i < replacements.length; i++) {
+        trade_info += replacements[i].owner_user_name + "\n\xA0Owned book: " + replacements[i].owned_book_name + "\n\xA0Desired book: "
+            + replacements[i].target_book_name + "\n\xA0";
+    }
+    logger.log(trade_info);
+    var replacement_text = {trade_info_text : trade_info};
+    var template = handlebars.compile(accepted_email_html);
+    var custom_html = template(replacement_text);
+    logger.log(emails);
+    sgMail.setApiKey(process.env.EMAIL_KEY);
```
Comment (codyleyhan) [inline models/emailer.js]: same here with the dep injection

**Comment by karenl1** [comment]
I only saw the accepted trade and rejected trade html files. Do we have all three email cases covered? (1) email to tell users when a trade has been matched, (2) when everyone has accepted a trade, and (3) when someone has rejected a trade. Also, can you attach screenshots of each type of email so we know what it looks like? 

After that, just merge master into your branch and we should be good. 😄

**Pull request merged**